### DISCUSSION TITLE
Selecting part of an equation

### DISCUSSION THREAD

**Discussion by ianchlee** [3b1b/manim#1313] [Q&A] [answered]
I have an equation written as `eq = TexMobject(r"\frac {x_1 x_2}{x_1 + x_2}")`. I want to select `x_1` for some manipulation. In particular, I want to color it without changing the other texts, and animate equation transformation, by moving `x_1` into its place at `eq2 = TexMobject(r"\frac{1}{x_1} + \frac{1}{x_2}")`. Could someone point me some directions on how to solve this? Many thanks!

**Comment by TonyCrane** [comment, marked as answer]
In your code, `x_1` in `eq` is `eq[0][:2]`, and in `eq2` is `eq2[0][2:4]`. The whole index-mob maps in `eq` and `eq2` are:
![7A2FE125-A594-408A-8E17-5BF1795DBA0F](https://user-images.githubusercontent.com/44120331/104466456-3a168b00-55f0-11eb-890e-4c3e966be4a2.jpeg)
![3D2E5F5A-E804-41C1-B544-7E0E3163386C](https://user-images.githubusercontent.com/44120331/104466472-3daa1200-55f0-11eb-83e8-018aa53b10e4.jpeg)
So you can try:
```python
self.play(TransformFromCopy(eq[0][:2], eq2[0][2:4]))
```

---
btw, if you want to learn more about `TexMobject` and it’s structure in manim, you can watch [this video](https://www.bilibili.com/video/BV1CC4y1H7kp). (Although this video is in Chinese, it has no great influence on learning the content. You can just watch the screen and ignore the Chinese subtitles.)

**Reply by ianchlee** [reply]
Thanks! This helped a lot!

**Comment by 3b1b** [comment]
The other way to do this would be to wrap the parts you want to be separated with two curly braces.  It's slightly awkward with fractions, since this functionality depends on the ordering of the parts of the string defining your Tex to match the ordering of symbols, so instead of using \frac{}{} you need to use \over.

`eq = Tex(r"{ {{x_1}} {{x_2}} \over {{x_1}}+{{x_2}}}")`

You could also write

`eq = Tex(r"x_1 x_2 \over x_1 + x_2", isolate=["x_1", "x_2"])`

Once it's been initialized like this, you can call things like `eq.get_parts_by_tex("x_1")` or `eq.set_color_by_tex("x_1", BLUE)`
